packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java

```diff
@@ -915,16 +915,17 @@ public ResponseEntity<TaskResponse> createEnrichDatasetTask(
 		);
 
 		// Grab the document
-		final Optional<DocumentAsset> document = documentAssetService.getAsset(documentId, permission);
-		if (document.isEmpty()) {
-			log.warn(String.format("Document %s not found", documentId));
-			throw new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("document.not-found"));
-		}
-
-		// make sure there is a text in the document
-		if (document.isPresent() && (document.get().getText() == null || document.get().getText().isEmpty())) {
-			log.warn(String.format("Document %s has no extracted text", documentId));
-			throw new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("document.extraction.not-done"));
+		DocumentAsset document = null;
+		if (documentId != null) {
+			document = documentAssetService
+				.getAsset(documentId, permission)
+				.orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("document.not-found")));
```

Comment:
this won't be a not found error... this would probably be a `service down` error